sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/BeamQueryPlanner.java

```diff
@@ -131,12 +139,42 @@ BeamRelNode convertToBeamRel(String sqlStatement)
               .replace(BeamLogicalConvention.INSTANCE)
               .replace(root.collation)
               .simplify();
-      beamRelNode = (BeamRelNode) planner.transform(0, desiredTraits, root.rel);
-      LOG.info("BeamSQL>\n" + RelOptUtil.toString(beamRelNode));
+
+      // original logical plan
+      RelNode originalRelNode = planner.transform(0, desiredTraits, root.rel);
+
+      // optimized logical plan
+      beamRelNode = optimizeLogicPlan(originalRelNode, desiredTraits);
+      LOG.info("OptimizedPlan>\n" + RelOptUtil.toString(beamRelNode));
     } finally {
       planner.close();
     }
-    return beamRelNode;
+    return (BeamRelNode) beamRelNode;
+  }
+
+  /** execute volcano planner. */
+  private RelNode runVolcanoPlanner(
+      RuleSet logicalOptRuleSet, RelNode relNode, RelTraitSet logicalOptTraitSet)
+      throws CannotPlanException {
+    Program optProgram = Programs.ofRules(logicalOptRuleSet);
+
+    try {
+      return optProgram.run(
+          relNode.getCluster().getPlanner(),
+          relNode,
+          logicalOptTraitSet,
+          Collections.EMPTY_LIST,
+          Collections.EMPTY_LIST);
+    } catch (CannotPlanException e) {
+      throw e;
+    }
+  }
+
+  private RelNode optimizeLogicPlan(RelNode relNode, RelTraitSet desiredTraits)
+      throws CannotPlanException {
+    RuleSet logicalOptRuleSet = BeamRuleSets.getRuleSets()[0];
```

Comment:
I was actually incorrect about how things work. It is nice to keep the sets of rules separate, but we probably do just want to put them all in the same bucket and run them together through our existing code paths.